mypyc/build.py

```diff
@@ -276,13 +283,57 @@ def write_file(path: str, contents: str) -> None:
             f.write(contents)
 
 
-def mypycify(paths: List[str],
-             mypy_options: Optional[List[str]] = None,
-             opt_level: str = '3',
-             multi_file: bool = False,
-             skip_cgen: bool = False,
-             verbose: bool = False,
-             strip_asserts: bool = False) -> List[Extension]:
+def construct_groups(
+    sources: List[BuildSource],
+    separate: Union[bool, List[Tuple[List[str], Optional[str]]]],
+    use_shared_lib: bool,
+) -> List[Tuple[List[BuildSource], Optional[str]]]:
+    if separate is True:
+        groups = [
+            ([source], None) for source in sources
+        ]  # type: List[Tuple[List[BuildSource], Optional[str]]]
+    elif isinstance(separate, list):
+        groups = []
+        for files, name in separate:
+            groups.append((
+                [src for src in sources if src.path in files],
+                name))
+    else:
+        groups = [(sources, None)]
+
+    # Generate missing names
+    groups = [
+        (group,
+         name or (
+             shared_lib_name([source.module for source in group]) if use_shared_lib else None))
+        for group, name in groups]
```

Comment:
I found this a bit hard to understand. It might be easier to read if it used for and if statements.